RetroBar/Controls/TaskList.xaml.cs

```diff
@@ -143,5 +143,24 @@ private void TasksScrollViewer_PreviewMouseWheel(object sender, System.Windows.I
                 e.Handled = true;
             }
         }
+
+        public bool AreAllWindowsMinimized()
```

Comment:
@dremin, is it possible that we need to bind AreAllWindowsMinimized to the ShowDesktop Checked state, or is there a better way to detect when all windows are minimized?

Reply:
I don't think you'll need to check if windows are minimized. From testing it seems like Explorer sets the desktop as the foreground window when "show desktop" is "enabled", and then considers it "enabled" until the desktop is no longer the foreground window. Once another window is foreground, toggling "show desktop" will show the desktop again rather than restore the windows. So basically once we enable show desktop, we want to keep the button checked until the foreground window changes.

There are a couple ways to do this, one being an approach similar to [FullScreenHelper](https://github.com/cairoshell/ManagedShell/blob/master/src/ManagedShell.AppBar/FullScreenHelper.cs) that polls `GetForegroundWindow()` every 100ms. Another idea that might be cleaner is to add an event to [TasksService](https://github.com/cairoshell/ManagedShell/blob/master/src/ManagedShell.WindowsTasks/TasksService.cs) on shell hook messages that RetroBar could add a handler for (maybe FullScreenHelper could even be refactored to use this down the road if appropriate).

Reply:
@xoascf Would you like me to help with the above to get this finished up?

Reply:
Oh that would be perfect, this week I tried to add it in ManagedShell but I don't quite understand how to do it as you described above, I would be very grateful if you could do it for me, maybe later changes can be made if needed or requested for some themes.

The "Allow edits..." option is enabled, you can make the changes you want in this PR or separately, @dremin.

Reply:
It seems that the behavior varies for certain themes:

In Windows 95 (and one would assume it was the same for classic pre-7 themes): The button appears pressed when all windows are minimized (either with this button itself or with a shortcut) and when the desktop is active.

In Windows 7 (and later): The button does not appear pressed when all windows are minimized or when it is pressed.

Reply:
@xoascf cairoshell/ManagedShell#74 is the ManagedShell support for this. Would you like me to wait to commit the RetroBar changes until a ManagedShell build is available on NuGet?

Reply:
> is the ManagedShell support for this. Would you like me to wait to commit the RetroBar changes until a ManagedShell build is available on NuGet?

That would be preferable.

Reply:
@xoascf Just pushed the changes.

Reply:
> @xoascf Just pushed the changes.


Thank you, by the way, the button is not selected when using the shortcut to display the desktop with the Win key + D. Is this intended?

Reply:
@xoascf Yeah, that is expected. I didn't add any handling for WM_COMMAND but will look into it. I don't think that should block this though.